server/src/main/java/org/elasticsearch/action/bulk/BulkShardRequest.java

```diff
@@ -143,6 +143,29 @@ public void writeTo(StreamOutput out) throws IOException {
         }
     }
 
+    @Override
+    public boolean supportsZeroCopy() {
```

Comment:
Not so sure about this, I kinda hate building logic based on `instanceof` but maybe just using a separate interface to give us the ability to do this would be better?